src/widgets/task-board/taskBoard.jsx

```diff
@@ -96,73 +130,158 @@ export function TaskBoard() {
+            name="in progress"
+            id="inProgress"
+            tasks={inProgressTasks}
+            onChangeStatus={handleDoneTask}
+            deleteTaskById={handleDeleteTaskById}
+            editTaskById={handleEditTaskById}
+            onDeleteTag={handleDeleteTagFromTask}
+          />
+          <TaskColumn
+            name="done"
+            id="done"
+            tasks={doneTasks}
+            deleteTaskById={handleDeleteTaskById}
+            editTaskById={handleEditTaskById}
+            onDeleteTag={handleDeleteTagFromTask}
+          />
         </Grid>
       </Box>
+
       <TaskForm
-        onClose={handleCloseForm}
-        open={isCreateForm}
-        onSubmit={handleCreate}
-        title="Create task"
-        btnText="Add task"
+        onClose={handleCloseTaskForm}
+        onButtonAddTag={handleClick}
+        buttonRef={buttonRef}
+        tags={selectedTagsByTaskId}
+        renderTagForm={({ onCheck, checkedTags }) => (
+          <TagList
+            open={openTagList}
+            anchorEl={anchorEl}
+            onClose={handleClose}
+            onOpenTagForm={handleOpenCreateTagForm}
+            tags={tags}
+            handleEditTagById={handleEditTagById}
+            onDeleteTagFromList={handleDeleteTagById}
+            onCheck={onCheck}
+            checkedTags={checkedTags}
+          />
+        )}
+        {...(isEditTaskForm
+          ? {
+              openTaskForm: isEditTaskForm,
+              onSubmit: handleEditTask,
+              text: selectedTask?.text,
+              title: 'Edit task',
+              btnText: 'Save',
+            }
+          : {
+              openTaskForm: isCreateTaskForm,
+              onSubmit: handleCreateTask,
+              title: 'Create task',
+              btnText: 'Add task',
+            })}
       />
-      <TaskForm
-        onClose={handleCloseForm}
-        open={isEditForm}
-        onSubmit={handleEdit}
-        text={selectedTask?.text}
-        title="Edit task"
-        btnText="Save"
+
+      <TagForm
+        onClose={handleCloseTagForm}
+        {...(isEditTagForm
+          ? {
+              onSubmit: handleEditTag,
+              open: isEditTagForm,
+              text: selectedTag?.name,
+              color: selectedTag?.color,
+              title: 'Edit tag',
+              btnText: 'Edit tag',
+            }
+          : {
+              onSubmit: handleAddTag,
+              open: isCreateTagForm,
+              title: 'Create tag',
+              btnText: 'Create tag',
+            })}
       />
+
       <ConfirmModal
         onClose={handleCloseConfirmModal}
-        onDelete={handleDeleteTask}
-        open={openConfirmModal}
+        {...(isTaskForm
+          ? {
+              children: 'Are you sure you want to delete the task?',
+              open: isTaskForm,
+              onDelete: handleDeleteTask,
+            }
+          : {
+              children: 'Are you sure you want to delete the tag?',
```

Comment:
Are you sure you want to delete **this** tag
Are you sure you want to delete **this** task